packages/components/src/sites-table/use-sites-table-sorting.tsx

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import type { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
```

Comment:
A late drive-by comment but is this a good idea? This imports data from the `calypso` package into the `@automattic/components` package, which is in turn imported into the `calypso` package, creating a circular dependency. It also adds a dependency to the `@automattic/components` package which is not part of the `dependencies` array in package.json.

I think this should be removed. If you need `SiteExcerptData` in this package, you should probably duplicate it here or move it to another package which `@automattic/components` can depend on.

Reply:
I've attempted a fix for this here: https://github.com/Automattic/wp-calypso/pull/66778

Reply:
Yes, very valid point.

We've already changed our other heavy-lifting hook (`useSitesTableFiltering()`) so that it operates on generic data (anything that looks sorta like a site), but forgot to do the same thing for this hook.
https://github.com/Automattic/wp-calypso/blob/2d6bf7416c800955578f831a3131fb1c407db6f1/packages/components/src/sites-table/use-sites-table-filtering.tsx#L45

I'll take a look at #66778